src/arrayMethodSplice.js

```diff
@@ -5,7 +5,78 @@
  */
 function applyCustomSplice() {
   [].__proto__.splice2 = function(start, deleteCount, ...items) {
-    // write code here
+    const deleteCountSplice = deleteCount;
+    const arrayItems = [...items];
+    const argumentsLength = arguments.length;
+    const sourceLength = this.length;
+    const returnedArray = [];
+    let startOfSplice = start;
+    // assign all constants for next work with them
+
+    // first situation with empty splice
+    if (argumentsLength === 0) return returnedArray;
+
+    // change start of splice if start < 0
+    if (startOfSplice < 0) {
+      startOfSplice += sourceLength;
+      if (startOfSplice < 0) { // checking if start is still less then 0
+        startOfSplice = Math.abs(startOfSplice);
+      }
+      if (startOfSplice > this.length) startOfSplice = 0;
+    }
+    let deletingLength = deleteCountSplice + Math.abs(startOfSplice);
+    // situation if it's just one argument in splice(already changed ^)
+    if (argumentsLength === 1) {
```

Comment:
will be better to check 
```javascript
    if (!deleteCount) {
```

or create some booleans variables

It is duplicated you use named arguments and `arguments` variable.